Forwarded: no
Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1006531
From: Daniel Black <daniel@mariadb.org>
Subject: mariadb: FTBFS on hurd-i386: undefined reference to misc functions and files

requires https://github.com/MariaDB/server/pull/2893 as debian
explicit architectures aren't needed since dh_auto_configure handles
this.

If it works, upstream welcome.

Hurd string from uname -m, "SYSTEM processor: i686-AT386" in mariadb
output. And wiki reference https://en.wikipedia.org/wiki/Uname

--- a/cmake/build_configurations/mysql_release.cmake
+++ b/cmake/build_configurations/mysql_release.cmake
@@ -118,7 +118,10 @@ ELSEIF(DEB)
   SET(WITH_ZLIB system CACHE STRING "")
   SET(WITH_LIBWRAP ON)
   SET(HAVE_EMBEDDED_PRIVILEGE_CONTROL ON)
-  SET(PLUGIN_AUTH_SOCKET YES CACHE STRING "")
+  # No hurd implementation
+  IF(NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "i686-AT386")
+    SET(PLUGIN_AUTH_SOCKET YES CACHE STRING "")
+  ENDIF()
   SET(WITH_EMBEDDED_SERVER ON CACHE BOOL "")
   SET(WITH_PCRE system CACHE STRING "")
   SET(CLIENT_PLUGIN_ZSTD OFF)
